addons/account/models/account_bank_statement.py

```diff
@@ -253,6 +254,16 @@ def button_confirm_bank(self):
             if moves:
                 moves.filtered(lambda m: m.state != 'posted').post()
             statement.message_post(body=_('Statement %s confirmed, journal items were created.') % (statement.name,))
+            if statement.journal_id.type == 'bank':
+                # Attach report to the Bank statement
+                content, content_type = self.env.ref('account.action_report_account_statement').render_qweb_pdf(statement.id)
+                self.env['ir.attachment'].create({
+                    'name': statement.name and _("Bank Statement %s.pdf") % statement.name or _("Bank Statement.pdf"),
+                    'type': 'binary',
+                    'datas': base64.encodestring(content),
+                    'res_model': statement._name,
+                    'res_id': statement.id
+                })
```

Comment:
use 
```xml
            attachment="''(object.state == 'posted') and blablabla"
            attachment_use="True"
```
in the report definition instead of reimplementing this feature manually

Reply:
@qdp-odoo : According to the spec, on validating bank statement need to attach the statement report to the Bank Statement. So I think we have to do this from python side(button_confirm_bank() of bank.statement), right ?

Reply:
@avw-odoo 
can you please clarify my doubts related to new format asked on 10th July  ?
@qdp-odoo  
can you please get back on my reply to your comments ?https://github.com/odoo/odoo/pull/34659